File: robin/java/src/level2/study4/HIndex.java
package level2.study4;

import java.util.Arrays;

public class HIndex {
    public static int solution(int[] citations) {

        int[] result = Arrays.stream(citations).sorted().toArray();

        int n = result.length;

        for (int i = 0; i < n; i++) {

            if (i + 1 > result[(n - 1) - i]) {
                return i;
            }
        }

        return n;
    }

    public static void main(String[] args) {
        int[] citations = {3, 0, 6, 1, 5};
        System.out.println(solution(citations));
    }

}

File: robin/python/programmers/level2/study4/h_index.py
def solution(citations):
    result = sorted(citations, reverse=True)

    for i in range(len(result)):
        print(i+ 1, result[i])
        if i + 1 > result[i]:
            return i

    return len(result)


if __name__ == '__main__':
    citations = [3, 0, 6, 1, 5]
    print(solution(citations))
